pydantic/types.py

```diff
@@ -389,8 +388,13 @@ def __class_getitem__(cls, item: AnyType) -> AnyType:
 
         @classmethod
         def __get_pydantic_core_schema__(
```

Comment:
Would it make sense to use `__pydantic_core_schema__ = core_schema.json_schema(None)`? Or do we prefer it like this to defer the initialization cost.

Reply:
Feels like ideally we'd use something like `@cached_property`, maybe later.